CHANGES.rst

```diff
@@ -1,6 +1,196 @@
+* Add casting keyword to numeric array types. (#547)
+* The Traits examples are now part of the Traits package, and so are
+  contributed to ``etsdemo``. (#1275)
+* The Traits examples package now includes a beginner's tutorial. (#1061)
+
+
+Changes
+~~~~~~~
+
+* The ``NoDefaultSpecified`` constant (used as a default value for
+  the ``TraitType`` ``default_value`` argument) is now public, made
+  available from ``traits.api``. (#1384, #1380, #1378)
+* The deprecation of the ``TraitMap`` trait type has been reversed, because
+  there are existing uses of ``TraitMap`` that are hard to replace.
+  Nevertheless, it's still not recommended to use ``TraitMap`` in new code. Use
+  ``Map`` instead. (#1365)
+* An attempt to use ``PrefixList`` with an empty list, or ``PrefixMap`` or
+  ``Map`` with an empty dictionary, now raises ``ValueError``. As a result,
+  the default default value (which used to be ``None``) is always valid.
+  (#1351)
+* ``TraitListEvent`` arguments are now keyword only. (#1346)
+* It's no longer necessary to specify a trait comparison mode of
+  ``ComparisonMode.identity`` when using ``observe`` to observe items
+  in a ``List``, ``Dict`` or ``Set``. (#1165, #1328, #1240)
+
+
+Documentation
+~~~~~~~~~~~~~
+
+* Clarify that only the ``api`` modules should be used for imports. (#1387)
+* Update copyright header end years. (#1376)
+* Update contents of ``image_LICENSE.txt``. (#1362)
+* Remove mentions of the removed functions ``adapts`` and ``implements`` from
+  the examples and tutorial. (#1367)
+* Move Traits introduction description to ``index.rst``. (#1358)
+* Fix path to Enthought logo when building docset. (#1285)
+* Fix the ``trait_documenter`` extension to be less fragile. (#1247)
+* Various other minor improvements, typo fixes, and other documentation fixes.
+  (#1396, #1383, #1381, #1384, #1292, #1355, #1350, #1319, #1292)
+
+Deprecations
+~~~~~~~~~~~~
+
+* The ``Function`` and ``Method`` trait types are deprecated. Use
+  ``Callable`` or ``Instance`` instead. (#1399, #1397)
+* The ``edit`` parameter to ``configure_traits`` has been deprecated. (#1311)
+* The ``UnittestTools._catch_warnings`` function has been deprecated. (#1310)
+* The use of the ``CHECK_INTERFACES`` global variable for automated
+  interface checking has been deprecated. (#1231)
+
+
+Fixes
+~~~~~
+
+* Non-``TraitError`` exceptions raised during ``Tuple`` validation are now
+  propagated. Previously they were converted into ``TraitError``. (#1393)
+* Dynamic ``Range`` and ``Enum`` traits are now properly validated
+  when inside a container (for example ``Tuple`` or ``List``). Previously
+  no validation was performed. (#1388, #1392)
+* Remove the unused module-level constant ``traits.has_traits.EmptyList``.
+  (#1366)
+* Don't hard-code class names in ``__repr__`` implementations of
+  ``TraitListEvent``, ``TraitSetEvent`` and ``TraitDictEvent``. (#1335)
+* Don't notify on empty ``update``s of ``Dict`` traits. (#1308)
```

Comment:
```suggestion
* Don't notify on empty ``update``\s of ``Dict`` traits. (#1308)
```

Reply:
Ah. I suspect this needs to be ```  ``update``\ s ```

Reply:
Turns out that either works, but I'm not fond of the restructuredtext specification here; I'll use the `\ s` version.